### ISSUE TITLE
Add timeZone support for ScheduledSparkApplications

### ISSUE THREAD

**Issue by fanalin** [kubeflow/spark-operator#1712]
k8s cronjobs got the option to schedule them with a specific timezone (see https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.25/#cronjobspec-v1-batch). 

It would be great if the ScheduledSparkApplications can be configured with a timeZone as well.

**Comment by indreshverma** [comment]
Can anyone from community please respond on this we are waiting for this feature

**Comment by yuchaoran2011** [comment]
Not sure how much adoption there is for ScheduledSparkApplication. But if anyone is willing to contribute, it will be a nice addition @indreshverma @fanalin

**Comment by jacobsalway** [comment]
It's not the nicest user experience but the cron library used in the operator does [support specifying a timezone](https://github.com/robfig/cron/blob/master/parser.go#L95), so you can specify an application with a schedule like this `schedule: "TZ=Australia/Sydney 0 3 * * *"`.

**Comment by jacobsalway** [comment]
@indreshverma fair point. I shouldn't be suggesting relying on the internal functionality in the cron library when Kubernetes never officially supported it. I'll come up with a CRD change and implementation for this.
